### ISSUE TITLE
Export/Import saved items of a conference

### ISSUE THREAD

**Issue by zuphilip** [Wilm0r/giggity#358]
When using Giggity on multiple devices, I would like to export my saved items from one of them and import it in the other one. Is something like this already possible? For example is there a simple file with all saved items somewhere in a folder which then could be replaced by another version?

**Comment by Wilm0r** [comment]
Hrm, looks like I've not been looking at the bug queue for a while... Apologies. :(

Interestingly exactly this functionality existed in the past, but it relied on the very old intent-based ZXing QR scanner. It didn't seem to get used much so I deleted it somewhat recently.

Also a while ago I considered adding login options + selections storage/sharing to ggt.gaa.st but never really completed that. It may be overkill, though of course manual syncing back and forth between devices is no fun either.

One other thing I thought would've been quite nice for example (to me as a person who actually quite hates using mobile phones) is a desktop web UI to browse a schedule and make selections, then export the results to a phone. But one of the few things I hate more than using phones is web development. ;)

Anyway, thanks for the vote here. I wonder what it'd take to revive this. Not as easy without the ZXing QR scanner/generator unless I import a QR generator library which is kind of against my "keep Giggity's .apk <512KiB" habit..

Oh and no, the selections are in an SQLite database, not that easy to export even with filesystem access.

**Comment by zuphilip** [comment]
I have found ZXingLite https://github.com/jenly1314/ZXingLite, could this be an alternative for you here?

Manual syncing is okay, if one does the planing on one device and then at the conference needs the selection on another device. But sure, some online syncing would even be better. Could WebDAV storage help there? It is standardized and should be easy for everyone to get somewhere a web storage with WebDAV support (e.g. nextcloud hosted ones).

**Comment by blinry** [comment]
I'd also like an export functionality to export the saved items, just for backup purposes when re-installing Android! :)

**Comment by Wilm0r** [comment]
This is in 3.0 on F-Droid now, and probably in the Play Store in a few days.

Imports are done too, just open the ggt.gaa.st link however you prefer. For example I installed https://f-droid.org/packages/com.wchung.qrshare/ for quick QR share/generation and can scan those on another phone. Or use Android QuickShare, or email them, whatever. Or you could even share your export link on Mastodon or so as a list of recommended talks. The import interface is nice now, it shows what the selections are and lets you pick which ones to import (or click a button for all).

<img width="599" height="776" alt="Giggity import screen" src="https://github.com/user-attachments/assets/3027184c-3d0b-4217-8ea9-c13f9e781832" />

If someone wants a bulk export/import for switching phones that's a separate request, please open a new one for that :)

**Comment by blinry** [comment]
Oh neat, that's really cool, thanks for implementing this @Wilm0r!

Generating the sharable URL seems to work well! But when we open it on another device, Giggity just displays the event, without offering to import events. Can someone reproduce this? (I can also open a new issue for this!) We used version 3.0.

**Comment by Wilm0r** [comment]
Yeah, sadly one bug snuck in that I fixed only a few days ago .... I should probably just push out a 3.0.1 very soon, for just this one bugfix. :)

**Comment by Wilm0r** [comment]
f82f775 Give it a day or two for F-Droid to pick up. And I should do a Play Store release.

**Comment by Wilm0r** [comment]
@blinry If anybody has Giggity from the Play Store, 3.0.1 is there already and has this bug fixed. I hope F-Droid will pick it up within the next day or so too.

**Comment by blinry** [comment]
Tested, works great! :)
